### ISSUE TITLE
Suggestion: Webtorrent and downloads window

### ISSUE THREAD

**Issue by rafaelcastrocouto** [brave/brave-browser#23962]
If the screens is large enough there could be an iframe to the downloads window next to the webtorrent files window, like splitting the screen in the middle. And on android/smaller screens the downloads window could always come after the webtorrent.

The reason is cuz it's really odd to have two tabs to follow torrent downloads. Even a single download requires the user to have two tabs opened to check on the download details and if you go for multiple downloads it becomes a tab nightmare.

**Comment by anonghuser** [comment]
I don't think just a split window is enough of a solution. In my opinion, if you are going to have built-in torrent support it really needs to be fully integrated into the downloads page. 

The current extension-like implementation feels really half-baked. 
It downloads the whole torrent somewhere temporary then user has to pick which files to save separately. Aside from being unintuitive and weird from UX POV, this has several technical downsides that are really bad when large files are involved:
- unclear where the temporary storage is or if it is properly cleaned up in all cases, i.e. on crashes
- doubled total space requirements while running, for temporary + permanent copies
- downloading unneeded parts of the torrent wasting time and bandwidth

You need to keep the torrent tab open. Closing it interrupts the download of all individual files you selected to save, with no way to resume, losing all progress. That's inconsistent with how downloads usually work. There are no warnings about that either, you learn it "the hard way". 
In the download window, instead of the magnet or torrent file URL, the individual files are listed with some weird localhost URL as their source, which will not work for re-downloading the file later.
The cancel, pause, resume, retry buttons for individual files in the download window do not affect or relate to the torrent's status and the start/stop buttons in the web-torrent window.

**Comment by rafaelcastrocouto** [comment]
I agree that we need a properly integrated downloads section, imho all items mentioned by @anonghuser are very important and most deserve their own issue. My proposal is meant to be just an inital seed into making this integration a reality.

The split window is not enough but it would already help a lot, with very small coding required for the implementation. This kind of big implementations usually takes a year or two (take the new tab implementaton #353 for eg). That why I think this is a good temporary change that can even help bringing more attention to the integration details missing.

Also seems woth creating a full integration issue, tracking each feature and managing their progress.

**Comment by rafaelcastrocouto** [comment]
Since this issue already has a little compilation of webtorrent issues, I thought I should drop this here, I can open a new issue if this starts to get too broad.

The "save all files" button does not account for memory limit and, for large files, will drop the entire page and show a memory limit error message, forcing the user to re-download all the torrents content again.

**Comment by bsclifton** [comment]
Closing as `wontfix`.

WebTorrent as a feature has been removed with https://github.com/brave/brave-browser/issues/46416 which goes out with version 1.81 (releasing 5th of August, 2025)

See https://github.com/brave/brave-browser/issues/46430 for more information
